### ISSUE TITLE
Enhancement: YAML file support

### ISSUE THREAD

**Issue by nielskrijger** [mozilla/node-convict#236]
I've been using Convict since I started with Node.js and any backend project I worked on I've migrated towards it when possible. Just love this library!

Because my last projects all used OpenAPI (swagger) and Kubernetes I've been using YAML a lot. It made sense there to migrate the config files to YAML too.

The problem was `convict.load(...parsed yaml contents...)` does not have the same priority as `convict.loadFile(jsonfile)`. My solution was to convert the yaml files to json, store it in a tmp location and load it normally using `loadFile()`.

I was wondering; is there any interest in adding YAML support?

I can think of several reasons why you would and would not want to support this. 

If there's clear support I might invest some time in creating a PR.

**Comment by jaawerth** [comment]
Similarly, I like using [TOML](https://github.com/toml-lang/toml) format for my config files for better human writability than JSON, as well as the occasional YAML...

Rather than supporting specific alternate formats, what I think would be nice is either an `addFileType(extension, { parse })` method, or add a second argument to `loadFile` along the same lines as the optional second argument to nconf's `.use` method [as described here](https://github.com/indexzero/nconf#file).

__addFileType__
```javascript
const { safeLoad: parseYaml } = require('js-yaml');
const { parse: parseToml } = require('toml');

convict.addFileType('toml', { parse: parseToml });
convict.addFileType('yml', { parse: parseYaml });

convict.loadFile('./config/somestuff.yml');
convict.loadFile('./config/whyamiusingtwoformatsohwell.toml');
```
__loadFile second arg__
```javascript
const { safeLoad: parseYaml } = require('js-yaml');
const { parse: parseToml } = require('toml');

convict.loadFile('./config/somestuff.yml', { parse: parsesYaml });
convict.loadFile('./config/whyamiusingtwoformatsohwell.toml', parseToml);
```

**Comment by rocketspacer** [comment]
Is this project dead?

**Comment by madarche** [comment]
Let's not add every file format to convict, but as suggested by @jaawerth add pluggable parsers. PR welcome.

@rocketspacer: the project is not dead, it's just stable, security updates are taken care of, and the project is also complex so we need to think twice before adding anything to the code.

**Comment by rocketspacer** [comment]
Glad to hear that. This project is my most favorite when it comes to dealing with configurations. Really hoping to see improvements coming.

**Comment by madarche** [comment]
Fixed by #244
